configuration/postgres/postgres.go

```diff
@@ -0,0 +1,296 @@
+	conn, err := p.client.Acquire(ctx)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Error acquiring connection:", err)
+	}
+	defer conn.Release()
+
+	_, err = conn.Exec(context.Background(), channel)
+	if err != nil {
+		p.logger.Errorf("Error listening to channel:", err)
+		return
+	}
+
+	for {
+		notification, err := conn.Conn().WaitForNotification(ctx)
+		if err != nil {
+			p.logger.Errorf("Error waiting for notification:", err)
+			return
+		}
+		p.handleSubscribedChange(ctx, handler, notification, id)
+	}
+}
+
+func (p *ConfigurationStore) handleSubscribedChange(ctx context.Context, handler configuration.UpdateHandler, msg *pgconn.Notification, id string) {
+	defer func() {
+		if err := recover(); err != nil {
+			p.logger.Errorf("panic in handleSubscribedChange(）method and recovered: %s", err)
+		}
+	}()
+	payload := make(map[string]interface{})
+	err := json.Unmarshal([]byte(msg.Payload), &payload)
+	if err != nil {
+		p.logger.Errorf("Error in UnMarshall: ", err)
+		return
+	}
+
+	var key, value, version string
+	m := make(map[string]string)
+	// trigger should encapsulate the row in "data" field in the notification
+	row := reflect.ValueOf(payload["data"])
+	if row.Kind() == reflect.Map {
+		for _, k := range row.MapKeys() {
+			v := row.MapIndex(k)
+			strKey := k.Interface().(string)
+			switch strings.ToLower(strKey) {
+			case "key":
+				key = v.Interface().(string)
+			case "value":
+				value = v.Interface().(string)
+			case "version":
+				version = v.Interface().(string)
+			case "metadata":
+				a := v.Interface().(map[string]interface{})
+				for k, v := range a {
+					m[k] = v.(string)
+				}
+			}
+		}
+	}
+	e := &configuration.UpdateEvent{
+		Items: map[string]*configuration.Item{
+			key: {
+				Value:    value,
+				Version:  version,
+				Metadata: m,
+			},
+		},
+		ID: id,
+	}
+	err = handler(ctx, e)
+	if err != nil {
+		p.logger.Errorf("fail to call handler to notify event for configuration update subscribe: %s", err)
+	}
+}
+
+func parseMetadata(cmetadata configuration.Metadata) (metadata, error) {
+	m := metadata{}
+
+	if val, ok := cmetadata.Properties[connectionStringKey]; ok && val != "" {
+		m.connectionString = val
+	} else {
+		return m, fmt.Errorf(ErrorMissingConnectionString)
+	}
+
+	if tbl, ok := cmetadata.Properties[configtablekey]; ok && tbl != "" {
+		m.configTable = tbl
+	} else {
+		return m, fmt.Errorf(ErrorMissingTableName)
+	}
+
+	// configure maxTimeout if provided
+	if mxTimeout, ok := cmetadata.Properties[connMaxIdleTimeKey]; ok && mxTimeout != "" {
+		if t, err := time.ParseDuration(mxTimeout); err == nil {
+			m.maxIdleTime = t
+		} else {
+			return m, fmt.Errorf(ErrorMissinMaxTimeout)
+		}
+	}
+	return m, nil
+}
+
+func Connect(ctx context.Context, conn string) (*pgxpool.Pool, error) {
+	pool, err := pgxpool.Connect(ctx, conn)
+	if err != nil {
+		return nil, fmt.Errorf("postgres configuration store connection error : %s", err)
+	}
+	pingErr := pool.Ping(context.Background())
```

Comment:
Should probably use a context that is child of `ctx` and has a timeout

Reply:
https://github.com/dapr/components-contrib/blob/90ba8a441d6ccc8987acf115ca4adf1f49fa9fba/configuration/postgres/postgres.go#L108-L112